plugins/modem-manager/fu-plugin-modem-manager.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <libmm-glib.h>
+
+#include "fu-plugin-vfuncs.h"
+
+#include "fu-mm-device.h"
+
+/* we need a new runtime version to support at+qfastboot */
+#define FU_PLUGIN_MODEM_MANAGER_REQUIRED_VERSION	"1.9.1"
+
+struct FuPluginData {
+	MMManager	*manager;
+};
+
+static void
+fu_plugin_mm_device_add (FuPlugin *plugin, MMObject *modem)
+{
+	const gchar *object_path = mm_object_get_path (modem);
+	g_autoptr(FuMmDevice) dev = NULL;
+	if (fu_plugin_cache_lookup (plugin, object_path) != NULL)
+		return;
+	dev = fu_mm_device_new (mm_object_peek_modem (modem));
+	if (!fu_device_probe (FU_DEVICE (dev), NULL))
+		return;
+	fu_plugin_device_add (plugin, FU_DEVICE (dev));
+	fu_plugin_cache_add (plugin, object_path, dev);
+}
+
+static void
+fu_plugin_mm_device_added_cb (MMManager *manager, MMObject *modem, FuPlugin *plugin)
+{
+	fu_plugin_mm_device_add (plugin, modem);
+}
+
+static void
+fu_plugin_mm_device_removed_cb (MMManager *manager, MMObject *modem, FuPlugin *plugin)
+{
+	const gchar *object_path = mm_object_get_path (modem);
+	FuMmDevice *dev = fu_plugin_cache_lookup (plugin, object_path);
+	if (dev == NULL)
+		return;
+	g_debug ("removed modem: %s", mm_object_get_path (modem));
+	fu_plugin_cache_remove (plugin, object_path);
+	fu_plugin_device_remove (plugin, FU_DEVICE (dev));
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	g_autolist(GDBusObject) modems = NULL;
```

Comment:
Using `g_autolist` requires glib 2.56.  Can you avoid using it (we only have 2.45.8 as a project dep right now)?

Otherwise snap will at least need to come up to a newer glib.